Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,426 @@
+                    damage = Math.Max(damage + (Math.Max(MoleClamp * mixture.Temperature - HeatDamage, 0) * SMcomponent.MoleHeatPenalty / 150) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Power only starts affecting damage when it is above 5000
+                    damage = Math.Max(damage + (Math.Max(SMcomponent.Power - SupermatterComponent.PowerPenaltyThreshold, 0f)/500f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //Molar count only starts affecting damage when it is above 1800
+                    damage = Math.Max(damage + (Math.Max(mixture.TotalMoles - SupermatterComponent.MolePenaltyThreshold, 0f)/80f) * SupermatterComponent.DamageIncreaseMultiplier, 0f);
+
+                    //There might be a way to integrate healing and hurting via heat
+			        //healing damage
+                    if (mixture.TotalMoles < SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        //Only has a net positive effect when the temp is below 313.15, heals up to 2 damage. Psycologists increase this temp min by up to 45
+                        var heatcap = ((Atmospherics.T0C + SupermatterComponent.HeatPenaltyThreshold) + (45 * SMcomponent.PsyCoeff));
+                        damage = Math.Max(-(2 * (1 - (mixture.Temperature / heatcap))),-2);
+                    }
+
+                    //if there are space tiles next to SM
+                    //TODO: change moles out for checking if adjacent tiles exist
+                    foreach(var adjacent in _atmosphere.GetAdjacentTileMixtures(Xform.Coordinates))
+                    {
+                        if (adjacent.TotalMoles == 0)
+                        {
+                            float factor;
+
+                            factor = integrity switch
+                            {
+                                (<= 25) => 0.002f,
+                                (<= 55) and (> 25) => 0.005f,
+                                (<= 75) and (> 55) => 0.0009f,
+                                (<= 90) and (> 75) => 0.0005f,
+                                _ => 0,
+                            };
+
+                            damage = Math.Clamp((SMcomponent.Power * factor) * SupermatterComponent.DamageIncreaseMultiplier, 0f, SupermatterComponent.MaxSpaceExposureDamage);
+                        }
+                    }
+                }
+
+                //only take up to 1.8 damage per cycle with no lower limmit
+                damage = Math.Min(SMcomponent.DamageArchived + (SupermatterComponent.DamageHardcap * SupermatterComponent.ExplosionPoint), damage);
+                //damage to add to total
+                var damageDelta = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Blunt"), (Shared.FixedPoint.FixedPoint2)damage);
+                _damageable.TryChangeDamage(SMcomponent.Owner, damageDelta, true);
```

Comment:
I don't think supermatter damage should be actual `DamageComponent` damage tbh. For getting damaged by explosions/attacks/bullets you should listen to the respective events as the SM has custom logic for those.

Reply:
i think it should be actual damage. but it should probably have its own damage type (so stuff like emitter bolts can do more damage to it) and its own resistances, not just using blunt.